backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/InlineClasses.kt

```diff
@@ -233,14 +244,23 @@ internal object KotlinTypeInlineClassesSupport : InlineClassesSupport<ClassDescr
         else type.constructor.supertypes.asSequence().flatMap { computeFullErasure(it) }
     }
 
-    override fun getFqName(clazz: ClassDescriptor): FqNameUnsafe = clazz.fqNameUnsafe
     override fun hasInlineModifier(clazz: ClassDescriptor): Boolean = clazz.isInline
 
     override fun getNativePointedSuperclass(clazz: ClassDescriptor): ClassDescriptor? = clazz.getAllSuperClassifiers()
             .firstOrNull { it.fqNameUnsafe == InteropFqNames.nativePointed } as ClassDescriptor?
 
     override fun getInlinedClassUnderlyingType(clazz: ClassDescriptor): KotlinType =
             clazz.unsubstitutedPrimaryConstructor!!.valueParameters.single().type
+
+    override fun getPackageFqName(clazz: ClassDescriptor) =
+            clazz.classId?.packageFqName
```

Comment:
Isn't this suboptimal?

Reply:
Why? `ClassDescriptor` stores packageFqName as  immutable property.